source/extensions/transport_sockets/tls/ocsp/ocsp.cc

```diff
@@ -116,15 +120,31 @@ OcspResponseWrapper::OcspResponseWrapper(std::vector<uint8_t> der_response, Time
 // Though different issuers could produce certificates with the same serial
 // number, this is check is to prevent operator error and a collision in this
 // case is unlikely.
-bool OcspResponseWrapper::matchesCertificate(X509& cert) {
+bool OcspResponseWrapper::matchesCertificate(X509& cert) const {
   std::string cert_serial_number = CertUtility::getSerialNumberFromCertificate(cert);
   std::string resp_cert_serial_number = response_->response_->getCertSerialNumber();
   return resp_cert_serial_number == cert_serial_number;
 }
 
 bool OcspResponseWrapper::isExpired() {
+  if (is_expired_.load()) {
```

Comment:
I think caching `is_expired_` is probably more trouble than it's worth, since it only gains anything if it is already expired (uncommon case, hopefully). I'd just delete it and do the comparison every time.